mava/components/training/losses.py

```diff
@@ -28,6 +29,91 @@
 from mava.core_jax import SystemTrainer
 
 
+class DefaultValueLossFunction(Component):
+    def __init__(
+        self,
+        config: SimpleNamespace = SimpleNamespace(),
+    ):
+        """Component defines a DefaultValueLossFunction loss function.
+
+        Args:
+            config: SimpleNamespace.
+        """
+        self.config = config
+
+    def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
+        """Creates and stores the squared error value loss function.
+
+        Args:
+            trainer: SystemTrainer.
+
+        Returns:
+            None.
+        """
+
+        def squared_error_loss_fn(
+            value_error: jax.interpreters.ad.JVPTracer,
+        ) -> jax.interpreters.ad.JVPTracer:
+            """Default loss function"""
+            return value_error**2
+
+        trainer.store.value_loss_fn = squared_error_loss_fn
+
+    @staticmethod
+    def name() -> str:
+        return "value_loss"
+
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        BaseTrainerInit required to set up trainer.store.trainer_agents,
+        trainer.store.trainer_agent_net_keys and trainer.store.networks.
+
+        Returns:
+            List of required component classes.
+        """
+        return [
+            training.BaseTrainerInit
+        ]  # import from training to avoid partial dependency
+
+
+@dataclass
+class HuberValueLossFunctionConfig:
+    huber_delta: float = 1.0
+
+
+class HuberValueLossFunction(DefaultValueLossFunction):
```

Comment:
Let's drop "Function" in class name. Also, not sure why we need to inherit from the default/SE loss? I think what we should do is create an abstract loss, as done below with the `Loss` class, but then be more specific with the difference between these classes. Might be good to have a quick offline chat about this? 🙂

Reply:
I think this could be a great option too, @DriesSmit  what do you think?

Reply:
Sounds good to me :slightly_smiling_face: